src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -81,17 +81,21 @@ trait IterableOps[+A] extends Any {
   /** A view representing the elements of this collection. */
   def view: View[A] = View.fromIterator(iterator())
 
-  /** Given a collection factory `fi` for collections of type constructor `C`,
-    *  convert this collection to one of type `C[A]`. Example uses:
+  /** Given a collection factory `fi`, convert this collection to the appropriate
+    * representation for the current element type `A`. Example uses:
     *
     *      xs.to(List)
     *      xs.to(ArrayBuffer)
+    *      xs.to(BitSet) // for xs: Iterable[Int]
     */
-  def to[C[X] <: Iterable[X]](fi: FromIterable[C]): C[A @uncheckedVariance] =
+  def to[F <: FromIterableBase[A]](fi: F): F#To[A @uncheckedVariance] =
```

Comment:
I think that this kind of type projections is going to be removed from Dotty because they are unsound. Since in our case the `fi` parameter would always be a static `object` I think that using a path-dependent type would not suffer from the problems you are mentioning.